pkg/operator/v1helpers/helpers.go

```diff
@@ -21,6 +22,41 @@ import (
 	operatorv1 "github.com/openshift/api/operator/v1"
 )
 
+var _ OperatorClient = &NoOpOperatorClient{}
+
+type NoOpOperatorClient struct{}
+
+// NewNoOpOperatorClient creates a new operator client which no-ops on any of the
+// methods of the OperatorClient interface.
+//
+// USE WITH CAUTION! Some controllers might only report their state by setting
+// operator conditions, you better make sure that a controller has other ways
+// to report important situations (e.g. it has proper logging or it creates
+// events).
+func NewNoOpOperatorClient() *NoOpOperatorClient {
+	return &NoOpOperatorClient{}
+}
+
+func (c *NoOpOperatorClient) GetObjectMeta() (*metav1.ObjectMeta, error) {
+	return &metav1.ObjectMeta{}, nil
+}
+
+func (c *NoOpOperatorClient) Informer() cache.SharedIndexInformer {
+	return nil
+}
+
+func (c *NoOpOperatorClient) GetOperatorState() (*operatorv1.OperatorSpec, *operatorv1.OperatorStatus, string, error) {
+	return &operatorv1.OperatorSpec{}, &operatorv1.OperatorStatus{}, "", nil
+}
+
+func (c *NoOpOperatorClient) UpdateOperatorSpec(_ string, _ *operatorv1.OperatorSpec) (*operatorv1.OperatorSpec, string, error) {
+	return &operatorv1.OperatorSpec{}, "", nil
+}
+
+func (c *NoOpOperatorClient) UpdateOperatorStatus(_ string, _ *operatorv1.OperatorStatus) (*operatorv1.OperatorStatus, error) {
```

Comment:
I'd like to see this method print the attempted state change at info level if conditions are "good" and at warning level with events if conditions are "bad".

The control loops that accept this as an interface expect to report status via this function today.